feat(serverless): update concepts (#4150)
scaleway/docs-content: serverless/functions/concepts.mdx

## GB-s

Unit used to measure the resource consumption of a Function. It reflects the amount of memory consumed over time.

## JWT Token

JWT (JSON Web Token) is an access token you can create from the console or API to enable an application to access your private container. [Find out how to secure a Function](/serverless/functions/how-to/secure-a-function/#restrict-access-to-your-functions).

## Handler

A handler is a routine/function/method that processes specific events. Upon invoking your function, the handler is executed and returns an output. Refer to our [dedicated documentation](/serverless/functions/reference-content/functions-handlers/) for more information on the structure of a handler.

## Instance

A Serverless Function instance handles incoming requests based on factors like the request volume, min scale, and max scale parameters.

## Load balancing

The Serverless infrastructure manages incoming request traffic. In scenarios like sudden traffic spikes or load testing, resources are automatically scaled based on the max scale parameter to handle the load.

## Logging

> Serverless offers a built-in logging system based on cockpit to follow the activity of your ressources: [monitoring Serverless Functions](/serverless/functions/how-to/monitor-function/).

Reviewer: ```suggestion
Serverless offers a built-in logging system based on cockpit to follow the activity of your resources: [monitoring Serverless Functions](/serverless/functions/how-to/monitor-function/).
```

## Max scale

This parameter sets the maximum number of function instances. You should adjust it based on your function's traffic spikes, keeping in mind that you may wish to limit the max scale to manage costs effectively.

## Metrics

> Performances of your Serverless ressources are natively available: [monitoring Serverless Functions](/serverless/functions/how-to/monitor-function/)).

Reviewer: ```suggestion
Performances of your Serverless resources are natively available: [monitoring Serverless Functions](/serverless/functions/how-to/monitor-function/)).
```

## Queue trigger

A queue trigger is a mechanism that connects a function to a queue created with [Scaleway Queues](/serverless/messaging/concepts/#queues), and invokes the function automatically whenever a message is added to the queue.

For each message that is sent to a queue, the trigger reads the message and invokes the associated function with the message as the input parameter.
The function can then process the message and perform any required actions, such as updating a database or sending a notification.

## Rolling update

hen deploying a new version of a Serverless Function, a rolling update happens by default. This means that the new version of the service is gradually rolled out to your users without downtime.
Here is how it works:

* When a new version of your function is deployed, the platform automatically starts routing traffic to the new version incrementally, while still serving requests from the old version until the new one is fully deployed.

> * Once the new version is successfully running, we gradually shifts all traffic to it, ensuring zero downtime.

Reviewer: ```suggestion
* Once the new version is successfully running, we gradually shift all traffic to it, ensuring zero downtime.
```

* The old version is decommissioned once the new version is fully serving traffic.

> This process ensures a seamless update experience, minimizing disruption to users during deployments. If needed, you can also manage traffic splitting between versions during the update process, allowing you to test new versions with a subset of traffic before fully migrating to it.

Reviewer: ```suggestion
This process ensures a seamless update experience, minimizing user disruption during deployments. If needed, you can also manage traffic splitting between versions during the update process, allowing you to test new versions with a subset of traffic before fully migrating to it.
```

Serverless.com (Serverless Framework) is a tool that enables the deployment of serverless applications without having to manage Serverless Function's API call. Just write your configuration in a YAML and deploy, it handles everything.

## Serverless Function

> A Serverless Function are serverless, fully managed compute services that allow you to run small, stateless code snippets or functions in response to HTTP requests or events.

Reviewer 2: ```suggestion
Serverless Functionq are serverless, fully managed compute services that allow you to run small, stateless code snippets or functions in response to HTTP requests or events.
```

> A Serverless Function are serverless, fully managed compute services that allow you to run small, stateless code snippets or functions in response to HTTP requests or events.

Reviewer: ```suggestion
Serverless Functions are serverless, fully managed compute services that allow you to run small, stateless code snippets or functions in response to HTTP requests or events.
```

> These functions automatically scale based on demand and are designed to be lightweight, event-driven, and easily deployable without worrying about infrastructure management. Functions is built on top of Serverless Containers, meaning you can run your functions packaged in containers and have them scale efficiently.

Reviewer 2: ```suggestion
These functions automatically scale based on demand and are designed to be lightweight, event-driven, and easily deployable, eliminating the need to worry about infrastructure management. Functions is built on top of Serverless Containers, meaning you can run your functions packaged in containers and have them scale efficiently.
```

## Serverless Job

Serverless Jobs are similar to Serverless Functions but are better suited for running longer workloads. See [the comparaison between Serverless products](/serverless/functions/reference-content/difference-jobs-functions-containers) for more information.

## Stateless

Refers to a system or application that does not maintain any persistent state between executions. In a stateless environment, each request or operation is independent, and no information is retained from previous interactions.

This means that each request is treated as a new and isolated event, and there is no need for the system to remember previous states or data once a task is completed. Statelessness is commonly used in serverless architectures where each function execution is independent of others.

> To store informations you can use [Scaleway Object Storage](/storage/object/), [Scaleway Managed Databases](/managed-databases/postgresql-and-mysql/), [Scaleway Serverless Databases](/serverless/sql-databases/).

Reviewer 2: ```suggestion
To store data you can use [Scaleway Object Storage](/storage/object/), [Scaleway Managed Databases](/managed-databases/postgresql-and-mysql/), and [Scaleway Serverless Databases](/serverless/sql-databases/).
```
